packages/block-library/src/rich-image/rich-image/index.js

```diff
@@ -0,0 +1,357 @@
+/**
+ * External dependencies
+ */
+
+import classnames from 'classnames';
+import Cropper from 'react-easy-crop';
+
+/**
+ * WordPress dependencies
+ */
+
+import {
+	BlockControls,
+	__experimentalBlock as Block,
+} from '@wordpress/block-editor';
+import { Fragment, Component } from '@wordpress/element';
+import {
+	Toolbar,
+	ToolbarButton,
+	Icon,
+	Button,
+	Spinner,
+	withNotices,
+	RangeControl,
+	DropdownMenu,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import richImageRequest from './api';
+import {
+	RotateLeftIcon,
+	RotateRightIcon,
+	FlipHorizontalIcon,
+	FlipVerticalIcon,
+	CropIcon,
+	AspectIcon,
+} from './icon';
+
+const ROTATE_STEP = 90;
+const DEFAULT_CROP = {
+	unit: '%',
+	x: 25,
+	y: 25,
+	width: 50,
+	height: 50,
+};
+const MIN_ZOOM = 1;
+const MAX_ZOOM = 3;
+const ZOOM_STEP = 0.1;
+const POPOVER_PROPS = { position: 'bottom right' };
+
+class RichImage extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			isCrop: false,
+			inProgress: null,
+			imageSrc: null,
+			imageSize: { naturalHeight: 0, naturalWidth: 0 },
+			crop: null,
+			position: { x: 0, y: 0 },
+			zoom: 1,
+			aspect: 4 / 3,
+			isPortrait: false,
+		};
+
+		this.adjustImage = this.adjustImage.bind( this );
+		this.cropImage = this.cropImage.bind( this );
+	}
+
+	adjustImage( action, attrs ) {
+		const { setAttributes, attributes, noticeOperations } = this.props;
+		const { id } = attributes;
+
+		this.setState( { inProgress: action } );
+		noticeOperations.removeAllNotices();
+
+		richImageRequest( id, action, attrs )
+			.then( ( response ) => {
+				this.setState( { inProgress: null, isCrop: false } );
+
+				if ( response.mediaID && response.mediaID !== id ) {
+					setAttributes( {
+						id: response.mediaID,
+						url: response.url,
+					} );
+				}
+			} )
+			.catch( () => {
+				noticeOperations.createErrorNotice(
+					__(
+						'Unable to perform the image modification. Please check your media storage.'
+					)
+				);
+				this.setState( { inProgress: null, isCrop: false } );
+			} );
+	}
+
+	cropImage() {
+		const { crop } = this.state;
+
+		this.adjustImage( 'crop', {
+			cropX: crop.x,
+			cropY: crop.y,
+			cropWidth: crop.width,
+			cropHeight: crop.height,
+		} );
+	}
+
+	render() {
+		const {
+			isSelected,
+			attributes,
+			originalBlock: OriginalBlock,
+			noticeUI,
+		} = this.props;
+		const {
+			isCrop,
+			inProgress,
+			position,
+			zoom,
+			aspect,
+			imageSize,
+			isPortrait,
+		} = this.state;
+		const { url } = attributes;
+		const isEditing = ! isCrop && isSelected && url;
+
+		if ( ! isSelected ) {
```

Comment:
This is causing the block to rerender when it is selected, which is not a great user experience esp. with an image that needs to load.